Dev

Daivols6 opened the pull request:
Add Diplom Project

Daivols6 committed:
Add Tests for Diplom

Daivols6 committed:
Add Tests for Diplom. Saving Version

Daivols6 committed:
Add Tests for DipomProject

Daivols6 committed:
Add Allure and fix TestCase

Daivols6 committed:
Fixed test cases, added logging and screenshots

Daivols6 committed:
add yml for ci/cd

Daivols6 committed:
add pipelines

Daivols6 committed:
add new pipelines

Daivols6 committed:
fix name in .yml

Daivols6 committed:
Fix yml file

Daivols6 committed:
Add Test for Azure

Daivols6 committed:
fixed checking of elements on the page

Daivols6 committed:
Fix name for test

Daivols6 committed:
Fix name test

Daivols6 committed:
Add End2End test for country USA
Delete using

VladimirVolkovVD commented on Diplom/PageObject/RegistrationPage.cs:
```diff
@@ -72,6 +72,51 @@ public RegistrationPage EnterAddressforUser()
+        public RegistrationPage EnterAddressUSAforUser()
+        {
+            var select = new SelectElement(driver.FindElement(Country));
+            select.SelectByIndex(0);
```
почему 0?

Daivols6 replied:
Переписал этот момент, теперь в случае с США нет выбора страны, остается выбранная по умолчанию

VladimirVolkovVD commented on Diplom/PageObject/RegistrationPage.cs:
```diff
@@ -72,6 +72,51 @@ public RegistrationPage EnterAddressforUser()
+            logger.Info($"Choosing a country");
+            AllureHelper.ScreenShot();
+            var random = new Random();
+            var Digit = random.Next(1, 7);
```
почему рандом что это за цифры?

Daivols6 replied:
Вынес отдельно приватную переменную

 private readonly int Digit = new Random().Next(1, 7);
 
Сделал метод для выбора случайного штата по индексу
 
 new DropDown().SelectByIndexNumber(State, Digit);

VladimirVolkovVD commented on Diplom/PageObject/RegistrationPage.cs:
```diff
@@ -72,6 +72,51 @@ public RegistrationPage EnterAddressforUser()
+            logger.Info($"Getting a new user");
+            driver.FindElement(FirstName).SendKeys(user.FirstName);
+            logger.Info($"Input FirstName");
+            AllureHelper.ScreenShot();
```
зачем делать столько скриншотов?

Daivols6 replied:
Удали метод ScreenShot() во всем коде
Создал метод  ScreenShotIfTestFailed(), делает скрины, только если тест упал
Изначально было много скриншотов, для более подробного отчета в Allure

VladimirVolkovVD commented on Diplom/Test/E2E.cs:
```diff
@@ -27,8 +27,19 @@ public void End2EndCheckout()
             new Payment().ChoisePaymentByCheck()
                          .AgreePayment()
                          .CheckSuccessMessage();
+        }
```
пустая строка

Daivols6 replied:
Удалил все лишние пустые строки в проекте

VladimirVolkovD reviewed (changes requested):
поправить коментарии

Daivols6 committed:
Added comments to methods
Rewrote some methods
I broke the cases into blocks
Removed extra spaces
Deleted extra classes
Deleted the screenshots, now the screenshot is only when the case falls

Daivols6 committed:
add default for value

Daivols6 committed:
add fix parse Value

Daivols6 committed:
fix value for var wait

Daivols6 committed:
Fix testcase for Azure

Daivols6 committed:
Added the always copy option for files with the setting

Daivols6 committed:
fix piplenes

Daivols6 committed:
add task for Azure

Daivols6 committed:
fix .csproj

Daivols6 committed:
add runsettings

Daivols6 committed:
Fix azure and allure config

Daivols6 committed:
Corrected the test cases
Returned spaces
Added a description of the methods

Daivols6 committed:
Fix azure

Daivols6 committed:
Create README.md

Daivols6 committed:
update Microsoft.NET.Test.Sdk

Daivols6 committed:
Merge branch 'Dev' of https://github.com/Daivols6/SeleniumNunit into Dev

Daivols6 committed:
fix Microsoft.NET.Test.Sdk
